plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1373,6 +1383,10 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
                     .toArray(Type[]::new);
             switch (task.getContent()) {
                 case POSITION_DELETES:
+                    if (fullyDeletedFiles.containsKey(task.getReferencedDataFile().orElseThrow())) {
+                        continue;
```

Comment:
I'd rather leave it. I liked having the file system operations separate, rather than having to wrap this whole loop in a try/catch for the IOExceptions. But I can move it if you'd prefer